src/plugins/segmentio/__tests__/batched-dispatcher.test.ts

```diff
@@ -201,41 +204,39 @@ describe('Batching', () => {
           "headers": Object {
             "Content-Type": "text/plain",
           },
+          "keepalive": false,
           "method": "post",
         },
       ]
     `)
   })
 
   describe('on unload', () => {
-    let unloadHandler: Function | undefined = undefined
-
-    beforeEach(() => {
-      jest
-        .spyOn(window, 'addEventListener')
-        .mockImplementation((evt, handler) => {
-          if (evt === 'beforeunload') {
-            unloadHandler = handler as Function
-          }
-        })
-    })
-
     it('flushes the batch', async () => {
       const { dispatch } = batch(`https://api.segment.io`)
 
-      await dispatch(`https://api.segment.io/v1/t`, {
+      dispatch(`https://api.segment.io/v1/t`, {
         hello: 'world',
-      })
+      }).catch(console.error)
```

Comment:
Do we want errors in `dispatch` to actually fail the test? It looks like it would have before.

Reply:
This is to get rid of the eslint error "No unhandled promises".

The reason `await` has been removed is to deliberately test how well the batched dispatcher handles a lot of "pressure" at once (`await` would have been simulating events coming in one by one).

The test would still will fail because if `dispatch` fails to do its job, latter assertions won't pass either.